src/Utils.ts

```diff
@@ -53,34 +53,33 @@ export function elementToText(element: Node): string {
     commentNode.parentNode.removeChild(commentNode);
 
   let text = "";
-
   const allNodes = <Element[]>XPath.select("//*", element);
   for (const node of allNodes) {
-    const nodeText = (node.nodeValue || (node.firstChild && node.firstChild.nodeType === 3 && node.firstChild.nodeValue) || "")
-      + ((node.nextSibling && node.nextSibling.nodeType === 3 && node.nextSibling.nodeValue) || "");
+    let nodeValue = (node.nodeValue || (node.firstChild && node.firstChild.nodeType === NodeType.TEXT_NODE && node.firstChild.nodeValue) || '').trim();
```

Comment:
Could we avoid making this mutable?
Maybe we could extract that logic in a function somewhere?